### PR TITLE
Fix defining a constructor proc inside class

### PR DESCRIPTION
**Pull request by konqoro**

### CODE REVIEW

File: content/content/oop_macro.md [hunk 1]
```diff
@@ -137,16 +137,19 @@ macro class*(head, body: untyped): untyped =
   # var declarations will be turned into object fields
   var recList = newNimNode(nnkRecList)
 
+  # expected name of constructor
+  let ctorName = newIdentNode("new" & $typeName)
+
   # Iterate over the statements, adding `this: T`
   # to the parameters of functions
```
Comment (flaviut) [inline content/content/oop_macro.md]: Add something like "unless the function is a constructor" to the comment.

File: content/content/oop_macro.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ macro class*(head, body: untyped): untyped =
     # --------------------
     # Ident !"Animal"
     typeName = head
-    baseName = ident("RootObj")
+    baseName = newIdentNode("RootObj")
```
Comment (flaviut) [inline content/content/oop_macro.md]: Why was this changed? I don't really have a problem with the change, but I'm curious wrt what your reasoning is.

File: content/content/oop_macro.md [hunk 3]
```diff
@@ -215,11 +223,15 @@ animals.add(Cat(name: "Mitten", age: 10))
 for a in animals:
   echo a.vocalize()
   echo a.age_human_yrs()
+
+let r = newRabbit("Fluffy", 3)
+echo r.vocalize()
```
Comment (flaviut) [inline content/content/oop_macro.md]: Make sure you print the age too!

File: content/content/oop_macro.md [hunk 4]
```diff
@@ -206,6 +209,11 @@ class Dog of Animal:
 class Cat of Animal:
```
Comment (flaviut) [inline content/content/oop_macro.md]: Also, update the comment at https://github.com/flaviut/nim-by-example/pull/45/files#diff-cb8c0be3e9f135b42d69c85f323a317cR184

File: content/content/oop_macro.md (hunk 2 above)
Comment (konqoro) [inline content/content/oop_macro.md]: Only because I was keeping to versions of the file. 
However I was very confused with ident, cause there two of them (actually three):
proc ident(n: NimNode): NimIdent {..}
proc ident(name: string): NimNode {..}
and the last one, used here, is just an alias for newIdentNode...

File: content/content/oop_macro.md (hunk 4 above)
Comment (konqoro) [inline content/content/oop_macro.md]: There is no constructor proc for the Animal class, so I don't see anything to update

**Comment by flaviut** [comment]
Thank you, looks great!
